Remove Bolt as a dependency from Neo4j Server 2.3.

jakewins opened the pull request:
- Keep Netty as a dependency, as we use Nettys encryption utilties to
  create X.509 certificates

thobe commented:
This does not look very invasive. It actually looks pretty harmless to me. In fact, the changes here seem less risky than the ones in #5568, since they rely on us remembering the `to be removed entirely as a dependency in 2.3.1` comments in several pom files...

jakewins committed:
Remove Bolt as a 2.3 dependency

jakewins commented:
I've updated this PR to keep the Netty logging adapter in place, and I've moved where that logger adapter gets plugged in to be before the TLS certificates are generated. This gets rid of TLS-logging cruft that was going into console.log by mistake.

If this is green, vote we close the other PR and merge this guy? /cc @nawroth @thobe @davidegrohmann
